### ISSUE TITLE
Rename released module to Microsoft.PowerShell.PSReadline for next major version

### ISSUE THREAD

**Issue by kilasuit** [PowerShell/PSReadLine#3787]
### Prerequisites

- [X] Write a descriptive title.

### Description of the new feature/enhancement

This is something that is happening with other modules that started out as non-team work and have then jumped into the product, including PSResourceGet, ThreadJob and others.

This is to align with that expectation

cc @daxian-dbw @StevenBucher98

**Comment by daxian-dbw** [comment]
I don't think this is needed for 2.3 release. PowerShell has the built-in modules like "CimCmdlets" and "PSDiagnostics", and I don't think they will be renamed any time soon.

**Comment by kilasuit** [comment]
I was thinking more for a 3.0 release so that it has plenty of time to be communicated if it's decided that this is something that we want to ensure of consistency or not.
